provider/sourcegraph-search/index.ts

```diff
@@ -0,0 +1,131 @@
+import type {
+    AnnotationsParams,
+    AnnotationsResult,
+    CapabilitiesParams,
+    CapabilitiesResult,
+    Item,
+    ItemsParams,
+    ItemsResult,
+    Provider,
+    ProviderSettings,
+} from '@openctx/provider'
+import { SourcegraphGraphQLAPIClient } from './graphql'
+import { isError, searchForFileChunks } from './search'
+
+const graphqlClient = new SourcegraphGraphQLAPIClient()
+
+interface ItemsParamsProposed extends ItemsParams {
+    candidate?: Item
```

Comment:
@thenamankumar just calling out here that I made `candidate` just a single item. Simplifies code, matches how we do it in cody, and I think makes sense?